File: 8kyu/dna-to-rna-conversion/dna-to-rna-conversion.ts
export function DNAtoRNA(dna: string): string {
	const swap: Record<string, string> = {
		T: "U",
		U: "T",
	};

  return [...dna].map(char => swap[char] ? swap[char] : char).join('')
}


console.log(DNAtoRNA("GCAT"))

File: 8kyu/dna-to-rna-conversion/dna-to-rna-conversion.go
package main

import (
	"fmt"
)

// func DNAtoRNA(dna string) string {
//   rna := strings.Replace(dna, "T", "U", -1)
// 	return rna
// }

func DNAtoRNA(dna string) string {
	var res string

	for _, letter := range dna {
		if letter == 'T' {
			res += "U"
		}

		res += string(letter)

	}

	return res
}

func main() {
	fmt.Println(DNAtoRNA("GCAT"))
}
